src/routing.js

```diff
@@ -44,21 +44,32 @@ export const queryStringMiddleware = (history, {reduxPathname, routes}, config =
         const result = next(action);
         const state = store.getState();
         const currentPath = reduxPathname(state);
-        const routeConfig = routes[currentPath];
+        const routesConfig = !_.isArray(routes[currentPath]) ? [routes[currentPath]] : routes[currentPath];
 
-        if (routeConfig) {
-            const {listen, select, serialize} = routeConfig;
-            const qsState = select(state);
+        if (!_.isEmpty(routesConfig)) {
+            let queryString = [];
+            routesConfig.forEach((routeConfig) => {
+                if (routeConfig) {
+                    const {select, serialize} = routeConfig;
+                    const qsState = select(state);
 
-            if (qsState === null) {
-                return result;
-            }
-            let data = _pruneNullLeaves(qsState);
-            if (listen === action.type) {
-                if (serialize) {
-                    data = _applySerialization(data, serialize);
+                    if (qsState === null) {
+                        return result;
+                    }
+                    let data = _pruneNullLeaves(qsState);
+                    if (_.contains(routesConfig.map((route) => (route.listen)), action.type)) {
```

Comment:
`({route: {listen}}) => listen` seems a bit more idiomatic to me, but it's a personal thing.